src/library/components/signup/async_validation_mixin.js

```diff
@@ -29,9 +29,9 @@ var AsyncValidationMixin = function() {
         _asyncValidationPassed: false
       });
       if (this._asyncValidationTimeoutID) {
-        clearTimeout(this._asyncValidationTimeoutID);
+        window.clearTimeout(this._asyncValidationTimeoutID);
```

Comment:
If you set the `browser` option in JSHint, you can use `setTimeout/clearTimeout` without prefixing with `window`, which would result in a smaller set of changes. Some people prefer the window prefix as it makes it more explicit, however, so it's a matter of preference.

Reply:
Good to know. I do like using the window object for the explicit reason.  So to keep with that theme, I just pushed an update that forces window.location and window.history.  Making that change pointed out another good reason to force the use of the window object: consistency. Without this a developer can use location or window.location.  (Of course I wouldn't be surprised if there was some jshint option to forbid the use window.location) But a downside I suppose is that now jshint won't catch typos when writing window.location or window.history.